File: src/main/java/ru/timakden/chesspuzzle/Knight.java
package ru.timakden.chesspuzzle;

/**
 * Конь, перемещается на любое поле доски Г-образным или L-образным ходом.
 */
class Knight extends ChessPiece {
    Knight(int row, int column) {
        super(row, column);
    }

    @Override
    public String toString() {
        return "N";
    }

    public boolean canAttackAnotherChessPiece(ChessPiece anotherChessPiece) {
        int row = getRow();
        int column = getColumn();
        int anotherRow = anotherChessPiece.getRow();
        int anotherColumn = anotherChessPiece.getColumn();

        return ((Math.abs(row - anotherRow) == 2) && (Math.abs(column - anotherColumn) == 1)) ||
                ((Math.abs(row - anotherRow) == 1) && (Math.abs(column - anotherColumn) == 2));
    }
}

File: src/main/kotlin/ru/timakden/chesspuzzle/Knight.kt
package ru.timakden.chesspuzzle

/**
 * Конь, перемещается на любое поле доски Г-образным или L-образным ходом.
 */
class Knight(row: Int, column: Int) : ChessPiece(row, column) {
    override fun toString() = "N"

    override fun canAttackAnotherPiece(anotherPiece: ChessPiece): Boolean {
        return Math.abs(row - anotherPiece.row) == 2 && Math.abs(column - anotherPiece.column) == 1 ||
                Math.abs(row - anotherPiece.row) == 1 && Math.abs(column - anotherPiece.column) == 2
    }
}
